src/lib/containers/Uploader.tsx

```diff
@@ -0,0 +1,156 @@
+      this.inputOpenFileRef.current.click()
+    }
+  }
+
+  handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (event.target.files) {
+      this.handleFilesChanged(event.target.files)
+    }
+  }
+
+  finishedProcessingOneFile = () => {
+    const incrementedFilesUploadedCount = this.state.filesUploadedCount + 1
+    const isStillUploading = incrementedFilesUploadedCount < this.state.totalFilesToUploadCount
+    this.setState(
+      {
+        isUploading: isStillUploading,
+        filesUploadedCount: incrementedFilesUploadedCount,
+        successfullyUploaded: !isStillUploading
+      })
+  }
+
+  handleUploadError = (error: any) => {
+    this.finishedProcessingOneFile()
+    this.setState({ error: { error } })
+  }
+
+  handleFilesChanged = (selectorFiles: FileList) => {
+    this.setState(
+      {
+        error: undefined,
+        isUploading: true,
+        filesUploadedCount: 0,
+        totalFilesToUploadCount: selectorFiles.length,
+        successfullyUploaded: false,
+      })
+    Array.from(selectorFiles).forEach((file) => {
+      // check for existing filename in parent folder before upload (add new version if exists).
+      // note that the parent container (project/folder) is configurable.
+      const newFileEntity: FileEntity = {
+        parentId: this.props.parentContainerId,
+        name: file.name,
+        concreteType: 'org.sagebionetworks.repo.model.FileEntity',
+        dataFileHandleId: '',
+      }
+      lookupChildEntity(
+        {
+          entityName: file.name,
+          parentId: this.props.parentContainerId
+        },
+        this.props.token).then((entityId: EntityId) => {
+          // ok, found an entity of the same name.
+          getEntity(this.props.token, entityId.id).then((existingEntity: FileEntity) => {
+            if (existingEntity.concreteType === 'org.sagebionetworks.repo.model.FileEntity') {
+              this.updateEntityFile(existingEntity, file)
+            } else {
+              this.updateEntityFile(newFileEntity, file)
+            }
+          })
+        }).catch((error: any) => {
+          if (error.statusCode === 404) {
+            // great, it's a new file!
+            this.updateEntityFile(newFileEntity, file)
+          } else {
+            this.handleUploadError(error)
+          }
+        })
+    })
+  }
+
+  updateEntityFile = (fileEntity: FileEntity, file: File) => {
+    uploadFile(this.props.token, file.name, file).then((fileUploadComplete: FileUploadComplete) => {
+      const isCreate = fileEntity.dataFileHandleId === ''
+      fileEntity.dataFileHandleId = fileUploadComplete.fileHandleId
+      const createOrUpdate = isCreate ? createEntity : updateEntity
+      createOrUpdate(fileEntity, this.props.token).then(() => {
+        this.finishedProcessingOneFile()
+      })
+    }).catch((error: any) => {
+      this.handleUploadError(error)
+    })
+  }
+
+  render() {
+    return (
+      <div>
+        <input
```

Comment:
For full clarity I think this could have a comment explaining that the input field is used only functionally, the button acts as the real UI.